gcsio/src/main/java/com/google/cloud/hadoop/gcsio/StatisticsTrackingHttpRequestInitializer.java

```diff
@@ -56,28 +43,8 @@ public void initialize(HttpRequest request) throws IOException {
 
   /** Set HTTP request stats */
   private void setHttpRequestStats(HttpRequest request) {
-    switch (request.getRequestMethod()) {
-      case "DELETE":
-        increment(HTTP_DELETE_REQUEST);
-        break;
-      case "GET":
-        increment(HTTP_GET_REQUEST);
-        break;
-      case "HEAD":
-        increment(HTTP_HEAD_REQUEST);
-        break;
-      case "PATCH":
-        increment(HTTP_PATCH_REQUEST);
-        break;
-      case "POST":
-        increment(HTTP_POST_REQUEST);
-        break;
-      case "PUT":
-        increment(HTTP_PUT_REQUEST);
-        break;
-      default:
-        throw new IllegalStateException("Unknown request method: " + request.getRequestMethod());
-    }
+    String requestMethod = Ascii.toUpperCase(request.getRequestMethod());
+    increment("HTTP_" + requestMethod + "_REQUEST");
```

Comment:
nit: String.format might be bit faster.

Reply:
`String.format` is actually slower than string concatenation: https://stackoverflow.com/a/31061425/3227693